### PR TITLE
trace:  Use non-generic to replace newEvictedQueue in trace.start to reduce memory usage.

### PR DESCRIPTION
**Pull request by tttoad**
benchstat:
```
goos: darwin
goarch: arm64
pkg: go.opentelemetry.io/otel/sdk/trace
              │     old     │                 new                 │
              │   sec/op    │   sec/op     vs base                │
TraceStart-10   950.6n ± 1%   641.0n ± 0%  -32.57% (p=0.000 n=10)

              │     old     │                new                 │
              │    B/op     │    B/op     vs base                │
TraceStart-10   1040.0 ± 0%   704.0 ± 0%  -32.31% (p=0.000 n=10)

              │    old     │                new                 │
              │ allocs/op  │ allocs/op   vs base                │
TraceStart-10   20.00 ± 0%   14.00 ± 0%  -30.00% (p=0.000 n=10)
```

### CODE REVIEW

File: sdk/trace/evictedqueue.go [hunk 1]
```diff
@@ -29,6 +29,22 @@ func newEvictedQueue[T any](capacity int) evictedQueue[T] {
 	}
 }
 
+func newEvictedQueueEvent(capacity int) evictedQueue[Event] {
```
Comment (dmathieu) [inline sdk/trace/evictedqueue.go:32]: Shouldn't we remove the `newEvictedQueue[T]` method?

Comment (tttoad) [reply sdk/trace/evictedqueue.go:32]: This method is also used in unit tests. Can be used to help test other methods.

Comment (dmathieu) [reply sdk/trace/evictedqueue.go:32]: We can switch the tests to use the two new methods rather than the generic one.

Comment (tttoad) [reply sdk/trace/evictedqueue.go:21]: @dmathieu Done. PTAL.

**Review by dmathieu — APPROVED** [review]

File: sdk/trace/evictedqueue.go [hunk 2]
```diff
@@ -19,13 +18,19 @@ type evictedQueue[T any] struct {
 	logDropped   func()
 }
 
-func newEvictedQueue[T any](capacity int) evictedQueue[T] {
-	var tVal T
-	msg := fmt.Sprintf("limit reached: dropping trace %T", tVal)
```
Comment (MadVikingGod) [inline sdk/trace/evictedqueue.go:24]: The majority of this change is due to these lines. Local testing, removing these lines and leaving it as generic, results in about -300ns, -304B, and -4 Allocs. 

We (the sig) should check if this pattern is used elsewhere, as it's probably causing performance issues.

File: CHANGELOG.md [hunk 3]
```diff
@@ -27,6 +27,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
   - The exporter no longer exports the deprecated "otel.library.name" or "otel.library.version" attributes.
 - Upgrade `go.opentelemetry.io/otel/semconv/v1.25.0` to `go.opentelemetry.io/otel/semconv/v1.26.0` in `go.opentelemetry.io/otel/sdk/resource`. (#5490)
 - Upgrade `go.opentelemetry.io/otel/semconv/v1.25.0` to `go.opentelemetry.io/otel/semconv/v1.26.0` in `go.opentelemetry.io/otel/sdk/trace`. (#5490)
+- `Tracer.Start` in `go.opentelemetry.io/otel/sdk/trace` use non-generic to replace `newEvictedQueue` to reduce memory allocation. (#5497)
```
Comment (MrAlias) [inline CHANGELOG.md:30]:
```suggestion
- Use non-generic functions in the `Start` method of `"go.opentelemetry.io/otel/sdk/trace".Trace` to reduce memory allocation. (#5497)
```

**Review by MrAlias — APPROVED** [review]

**Pull request merged**